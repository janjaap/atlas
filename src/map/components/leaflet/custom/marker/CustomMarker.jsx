import { Marker } from 'react-leaflet';

class CustomMarker extends Marker {
  updateLeafletElement(fromProps, toProps) {
    super.updateLeafletElement(fromProps, toProps);
    if (toProps.rotationAngle !== fromProps.rotationAngle) {
      this.leafletElement.setRotationAngle(toProps.rotationAngle);
    }
  }
}

export default CustomMarker;
