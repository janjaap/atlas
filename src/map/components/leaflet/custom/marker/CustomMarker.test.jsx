import React from 'react';
import { shallow } from 'enzyme';
import CustomMarker from './CustomMarker';

describe('CustomMarker', () => {
  it('should render everything', () => {
    const wrapper = shallow(
      <CustomMarker
        position={[]}
        zIndexOffset={100}
        rotationAngle={0}
      />, {
        context: {
          layerContainer: {
            addLayer: jest.fn()
          }
        }
      }
    );
    const setRotationAngleMock = jest.fn();

    wrapper.instance().leafletElement = {
      setRotationAngle: setRotationAngleMock
    };
    wrapper.instance().updateLeafletElement({ rotationAngle: 0 }, { rotationAngle: 1 });

    expect(wrapper).toMatchSnapshot();
    expect(setRotationAngleMock).toHaveBeenCalled();
  });
});
