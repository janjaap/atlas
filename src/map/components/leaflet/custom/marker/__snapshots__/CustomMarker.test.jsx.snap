// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMarker should render everything 1`] = `""`;
